File: framework/languages/node/samples/Bingo.Ts/Server/Play/Infrastructure/ZLink/Spots/BingoRoomSpot/Handlers/bingo-reward-acquired-event-handler.ts
import { Injectable } from '@nestjs/common';
import { zlinkSpotSubscriptionHandler } from '@zlink-systems/nestjs';
import { SampleNames } from '../../../../../../Configuration/sample-names';
import { BingoRoomSpot } from '../bingo-room-spot';
import type { ZLinkPublishMessageContext, ZLinkSpotSubscriptionHandler } from '@zlink-systems/framework';
import type { BingoRewardAcquiredEvent } from '../../../../../../../Shared/Contracts/messages';

@Injectable()
@zlinkSpotSubscriptionHandler({
  spot: () => BingoRoomSpot,
  channelName: SampleNames.roomRewardChannel,
  topic: SampleNames.roomRewardTopic
})
class BingoRewardAcquiredEventHandler
  implements ZLinkSpotSubscriptionHandler<BingoRoomSpot, BingoRewardAcquiredEvent> {
  async handle(
    room: BingoRoomSpot,
    event: BingoRewardAcquiredEvent,
    context: ZLinkPublishMessageContext
  ): Promise<void> {
    void context;
    await room.announceReward(event);
  }
}

export { BingoRewardAcquiredEventHandler };

File: framework/languages/dotnet/samples/Bingo/Server/Play/Infrastructure/ZLink/Spots/BingoRoomSpot/Handlers/BingoRewardAcquiredEventHandler.cs
using Bingo.Server.Configuration;
using Bingo.Shared.Contracts;
using Zlink.Framework.Contracts.Channels;
using Zlink.Framework.Contracts.Handlers;
using Zlink.Framework.Contracts.Spots;

namespace Bingo.Server.Play.Infrastructure.ZLink.Spots.BingoRoomSpot.Handlers;

[ZLinkSpotSubscriptionHandler(SampleNames.RoomChannel, SampleNames.RewardTopic)]
internal sealed class
    BingoRewardAcquiredEventHandler : IZLinkSpotSubscriptionHandler<BingoRoom, BingoRewardAcquiredEvent>
{
    public ValueTask HandleAsync(
        BingoRoom spot,
        BingoRewardAcquiredEvent message,
        ZLinkPublishMessageContext context,
        CancellationToken cancellationToken)
    {
        return spot.AnnounceRewardAsync(message, cancellationToken);
    }
}
